sd_bandits/obp_extensions/dataset.py

```diff
@@ -188,4 +327,142 @@ def obtain_batch_bandit_feedback(
+        policy, where users click actions based on the "ground-truth" click probabilities calculated
+        from the user/playlist feature dot-product.
+
+        If cascade is enabled, we don't observe any rewards after max(len_init, last_click_index).
+
+        Parameters
+        ----------
+        policy : Union[BaseContextFreePolicy, BaseContextualPolicy]
+            Policy that will be used for online simulation.
+        n_batches : int
+            How many batches of sampled users to simulate.
+        users_per_batch : int
+            How many users to sample each batch.
+        seed : int
+            Random seed.
+        cascade : bool
+            Whether to simulate cascade logic, where rewards after max(len_init, last_click_index)
+            are not observed.
+        cascade_at : str
+            One of "first" or "last" indicating if we want to stop observing at the first or
+            last clicked item.
+
+        Returns
+        -------
+        dict
+            A feedback dict containing:
+                action : np.array (n_rounds,)
+                    Array of actions presented.
+                reward : np.array (n_rounds,)
+                    Array of rewards returned.
+                position : np.array (n_rounds,)
+                    Array of the position of each action.
+                context : np.array (n_rounds, 97)
+                    Array of the user context for each round.
+                action_context : np.array (n_rounds, 97)
+                    Array of the action context for each round.
+                pscore: NoneType
+                    N/A.
+                n_rounds : int
+                    Number of users per batch * number of batches * number of items observed for that
+                    user's session.
+                n_actions : int
+                    Number of available items (always 862).
+                users : np.array (n_rounds * users_per_round,)
+                    The users that were generated for this simulation.
+
+        """
+        policy.batch_size = np.inf
+        rng = np.random.default_rng(seed)
+        user_indices = rng.choice(
+            range(len(self.user_features)),
+            size=users_per_batch * n_batches,
+            replace=True,
+        )
+
+        all_probabilities = []
+        all_item_indices = []
+
+        relevant_user_features = self.user_features[user_indices, :]
+
+        actions = []
+        rewards = []
+        positions = []
+        context = []
+        selected_actions = []
+
+        for i, user_idx in tqdm(
+            enumerate(user_indices),
+            desc="Simulating online learning",
+            total=len(user_indices),
+        ):
+            if policy.policy_type == "contextfree":
+                item_indices = policy.select_action()
+            else:
+                item_indices = policy.select_action(self.user_features[user_idx])
```

Comment:
Here is where we can add an elif block when doing segments
`elif policy.policy_type == "segmented":
 item_indices =  policy.select_action(self.user_segment[user_idx])`